File: code/py/src/luogu/contest/c20231009/qF_p1332.py
def main() -> None:
    n, m, a, b = map(int, input().split())
    source_ls = []
    for _ in range(a):
        x, y = map(int, input().split())
        source_ls.append((x, y))
    MIN_VAL = n + m - 2
    for _ in range(b):
        x, y = map(int, input().split())
        min_val = MIN_VAL
        for source_x, source_y in source_ls:
            min_val = min(min_val, abs(x - source_x) + abs(y - source_y))
        print(min_val)


if __name__ == '__main__':
    main()

File: code/c/src/onlinejudge/luogu/contest/c20231009njit/qF_p1332.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    int n, m, a, b;
    scanf("%d %d %d %d", &n, &m, &a, &b);
    int sourceList[100001][2];
    for (int i = 0; i < a; i++) {
        int x, y;
        scanf("%d %d", &x, &y);
        sourceList[i][0] = x;
        sourceList[i][1] = y;
    }
    const int MIN_VAL = n + m - 2;
    for (int i = 0; i < b; i++) {
        int x, y;
        scanf("%d %d", &x, &y);
        int minVal = MIN_VAL;
        for (int j = 0; j < a; j++) {
            int sourceX = sourceList[j][0];
            int sourceY = sourceList[j][1];
            int distance = abs(x - sourceX) + abs(y - sourceY);
            if (distance < minVal) {
                minVal = distance;
            }
        }
        printf("%d\n", minVal);
    }
    return 0;
}